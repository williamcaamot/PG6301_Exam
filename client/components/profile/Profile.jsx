import React, {useContext} from "react";
import {AppContext} from "../App.jsx";

function Profile() {


    const {user} = useContext(AppContext);

    return <>
        <div className={"pageContentWrapper"}>
            <div className={"innerWrapper"}>
                <div style={{width:"100%"}}>
                    <h2>{user && user.name} {user && user.family_name}</h2>
                </div>
                <div style={{width:"100%"}}>
                    <p>E-post: {user && user.email}</p>
                </div>
                <div style={{width:"100%"}}>
                    <p>Kallenavn</p>
                </div>
                <div style={{width:"100%"}}>
                    <p>Bio:</p>
                </div>
            </div>
        </div>
    </>
}

export default Profile;