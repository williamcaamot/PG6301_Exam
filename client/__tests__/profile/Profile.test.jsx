import renderer, { act } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Profile from "../../components/profile/Profile.jsx";

describe("show user profile details", () => {
  it("render a user profile with information", () => {
    const user = {
      _id: "655dd38a4c15f00c20bc3fcf",
      bio: "dawawddawJeg er kul",
      email: "williamcoucheronaamot@gmail.com",
      family_name: "Coucheron-Aamot",
      name: "William",
      nickname: "WILLIAM123wwwww",
      picture:
        "https://lh3.googleusercontent.com/a/ACg8ocIOgSmXkWTJqoKPe5Hk3JlkkhfUNer4sWkVo3UuIa_dzg=s96-c",
    };

    const component = renderer.create(
      <MemoryRouter>
        <Profile user={user} />
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
