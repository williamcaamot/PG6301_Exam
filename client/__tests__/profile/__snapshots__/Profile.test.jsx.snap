// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show user profile details render a user profile with information 1`] = `
<div
  className="userProfile"
>
  <img
    alt="User Profile"
    src="https://lh3.googleusercontent.com/a/ACg8ocIOgSmXkWTJqoKPe5Hk3JlkkhfUNer4sWkVo3UuIa_dzg=s96-c"
  />
  <p>
    Kallenavn: 
    WILLIAM123wwwww
     
  </p>
  <p>
    Fult navn: 
    William
     
    Coucheron-Aamot
  </p>
  <p>
    E-post: 
    williamcoucheronaamot@gmail.com
  </p>
  <p>
    Om meg: 
    dawawddawJeg er kul
  </p>
</div>
`;
